packages/api/cms-api/src/common/helper/partial-type.helper.ts

```diff
@@ -0,0 +1,34 @@
+import { Type } from "@nestjs/common";
+import { inheritPropertyInitializers, inheritTransformationMetadata, inheritValidationMetadata, MappedType } from "@nestjs/mapped-types";
+
+import { IsUndefinable } from "../validators/is-undefinable";
+
+//Copy from @nestjs/mapped-types with applyIsOptionalDecorator changed to use IsUndefinable instead of IsOptional
+export function PartialType<T>(classRef: Type<T>): MappedType<Partial<T>> {
```

Comment:
probably not a good idea to use the exact same name as nestjs does, but I couldn't come up with a good name.

Reply:
Two suggestions
- PartialUpdateInputType
- PartialUpdateType

Reply:
But I'm okay with PartialType too

Reply:
I'd use PartialType as this does not more or less as described in https://docs.nestjs.com/openapi/mapped-types but better and customized for our needs.